### PR TITLE
Use OkHttp's certificate creation code

### PR DESCRIPTION
**Pull request by swankjesse**
We don't implement the full feature set that Bouncycastle has, but
we also don't need it.

In follow up changes I intend to remove the Bouncycastle dependency
for everything but some test cases.

### CODE REVIEW

File: okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt [hunk 1]
```diff
@@ -315,88 +321,134 @@ class HeldCertificate(
...
-        val encodableAltNames = arrayOfNulls<ASN1Encodable>(altNames.size)
-        for (i in 0 until altNames.size) {
-          val altName = altNames[i]
-          val tag = when {
-            altName.canParseAsIpAddress() -> GeneralName.iPAddress
-            else -> GeneralName.dNSName
+        val extensionValue = altNames.map {
+          when {
+            it.canParseAsIpAddress() -> {
+              generalNameIpAddress to InetAddress.getByName(it).address.toByteString()
```
Comment (yschimke) [inline okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt:424]: Is this known to be expanded to the full form always for IPv6?

Comment (swankjesse) [reply okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt:424]: Yep. The result will always be 4 or 16 bytes.

File: okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt [hunk 2]
```diff
@@ -315,88 +321,134 @@ class HeldCertificate(
...
-
-      // Subject, public & private keys for this certificate's signer. It may be self signed!
-      val signedByKeyPair: KeyPair
-      val signedByPrincipal: X500Principal
+      // Subject keys & identity.
+      val subjectKeyPair = keyPair ?: generateKeyPair()
+      val subjectPublicKeyInfo = CertificateAdapters.subjectPublicKeyInfo.fromDer(
+          subjectKeyPair.public.encoded.toByteString()
+      )
+      val subject: List<List<AttributeTypeAndValue>> = subject()
```
Comment (oldergod) [inline okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt:329]: woof

File: okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt [hunk 3]
```diff
@@ -315,88 +321,134 @@ class HeldCertificate(
...
+          )
+      )
+
+      return HeldCertificate(subjectKeyPair, certificate.toX509Certificate())
+    }
+
+    private fun subject(): List<List<AttributeTypeAndValue>> {
+      val result = mutableListOf<List<AttributeTypeAndValue>>()
+
+      if (ou != null) {
```
Comment (oldergod) [inline okhttp-tls/src/main/kotlin/okhttp3/tls/HeldCertificate.kt:382]: There's no better name for those?

**Pull request merged**